src/plugins/data/common/index_patterns/index_patterns/index_pattern.test.ts

```diff
@@ -97,16 +97,18 @@ const apiClient = {
 
 // helper function to create index patterns
 function create(id: string, payload?: any): Promise<IndexPattern> {
-  const indexPattern = new IndexPattern(
-    id,
-    (cfg: any) => config.get(cfg),
-    savedObjectsClient as any,
+  const indexPattern = new IndexPattern(id, {
+    getConfig: (cfg: any) => config.get(cfg),
+    savedObjectsClient: savedObjectsClient as any,
```

Comment:
Since you're already in this code could we get rid of these `as any` references? Seems like it should be fairly trivial to actually use the `savedObjectsServiceMock.createStartContract()` and then do something like `savedObjectsClient.get.mockReturnValue(object)`, etc.

Reply:
I took a look at this but realized that these dependencies will be changing shortly. This is code in `common` but we don't have common mocks. 

I improved the fieldForms mock as that is the same across `public` and `server`